client/src/components/Admin/order.controller/OrderController.tsx

```diff
@@ -1,47 +1,49 @@
 import axios from "axios";
-import { useState, useEffect } from "react";
+import React, { useState, useEffect, FC } from "react";
 import { useParams, useHistory} from "react-router-dom"
 import '../order.controller/order-update-form.css'
-
+import { Params, User, Clock, City, Master } from '../../../types/types'
 
 const currentDate = new Date() 
 const currentDay = (currentDate.getDate() < 10) ? `0${currentDate.getDate()}` : currentDate.getDate()
 const currentMonth = ((currentDate.getMonth() + 1) < 10) ? `0${(currentDate.getMonth() + 1)}` : (currentDate.getMonth() + 1)
 const currentYear = currentDate.getFullYear()
 const today = `${currentYear}-${currentMonth}-${currentDay}`
 
-const openingHours = ['09:00', '10:00', '11:00', '12:00', '13:00', '14:00', '15:00', '16:00', '17:00', '18:00']
+const openingHours: string[] = ['09:00', '10:00', '11:00', '12:00', '13:00', '14:00', '15:00', '16:00', '17:00', '18:00']
+
+interface OrderControllerProps {}
 
-const OrderController = () => {
+const OrderController: FC<OrderControllerProps> = () => {
 
     const history = useHistory()
 
-    const { propsOrderId, propsUserId, propsClockId, propsCityId, propsOrderDate, propsOrderTime, propsMasterId } = useParams()
+    const { propsOrderId, propsUserId, propsClockId, propsCityId, propsOrderDate, propsOrderTime, propsMasterId } = useParams<Params>()
 
-    const [userId, setUserId] = useState(0)
-    const [users, setUsers] = useState([])
+    const [userId, setUserId] = useState<number>(0)
+    const [users, setUsers] = useState<User[]>([])
 
-    const [clockId, setClockId] = useState(0)
-    const [clocks, setClocks] = useState([])
+    const [clockId, setClockId] = useState<number>(0)
+    const [clocks, setClocks] = useState<Clock[]>([])
 
-    const [cityId, setCityId] = useState(0)
-    const [cities, setCities] = useState([])
+    const [cityId, setCityId] = useState<number>(0)
+    const [cities, setCities] = useState<City[]>([])
 
-    const [orderDate, setOrderDate] = useState(`${propsOrderDate}`)
-    const [orderTime, setOrderTime] = useState(`${propsOrderTime}`)
+    const [orderDate, setOrderDate] = useState<string>(propsOrderDate)
+    const [orderTime, setOrderTime] = useState<string>(propsOrderTime)
 
-    const [masterId, setMasterId] = useState(0)
-    const [masters, setMasters] = useState([])
+    const [masterId, setMasterId] = useState<number>(0)
+    const [masters, setMasters] = useState<Master[]>([])
 
 
     useEffect(() => {
         
         const readAllUsers = async () => {
 
-            const {data} = await axios.get(`/user`)
+            const {data} = await axios.get<User[]>(`/user`)
 
             setUsers(data)
-            setUserId(propsUserId)
+            setUserId(+propsUserId)
```

Comment:
Why do you need plus sign here?

Reply:
> 

TS complaine about type string of 'propsUserId' that useParams returns

Reply:
@oleshkoo-clockwise  TS complaine about type string of 'propsUserId' that useParams returns